src/Mod/Part/App/PartFeature.cpp

```diff
@@ -483,13 +483,12 @@ static std::vector<std::pair<long, Data::MappedName>> getElementSource(App::Docu
                     break;
                 }
             }
-            // TODO: 02/24 Toponaming project:  It appears that getElementOwner is always nullptr.
-            //            if (owner->isDerivedFrom(App::GeoFeature::getClassTypeId())) {
-            //                auto o =
-            //                static_cast<App::GeoFeature*>(owner)->getElementOwner(ret.back().second);
-            //                if (o)
-            //                    doc = o->getDocument();
-            //            }
+            if (owner->isDerivedFrom(App::GeoFeature::getClassTypeId())) {
+                auto o =
+                static_cast<App::GeoFeature*>(owner)->getElementOwner(ret.back().second);
+                if (o)
+                    doc = o->getDocument();
+            }

```

Comment:
This probably can be rewritten as (not sure how the formatting should look in that case):
```cpp
if (
    owner->isDerivedFrom(App::GeoFeature::getClassTypeId()) && 
    auto ownerGeoFeature = static_cast<App::GeoFeature*>(owner)->getElementOwner(ret.back().second)
) {
    doc = ownerGeoFeature->getDocument();
}
```

Reply:
You are right.  Thanks for looking at this with fresh eyes and leaving a review.